bundles/org.openhab.binding.miele/src/main/java/org/openhab/binding/miele/internal/discovery/MieleMDNSDiscoveryParticipant.java

```diff
@@ -124,12 +125,13 @@ public String getServiceType() {
         try {
             socket = new Socket(addresses[0], 80);
             InetAddress ourAddress = socket.getLocalAddress();
-            String interfaceIpAddress = ourAddress.getHostAddress();
+            NetworkInterface ni = NetworkInterface.getByInetAddress(ourAddress);
+            String interfaceName = ni.getName();
```

Comment:
This is quite the hack, setting up a connection just to figure out which network interface to use. The method added to `NetUtil` here could be a less hacky way to do it, although it wouldn't work if the destination network is routed (not on a local subnet): https://github.com/openhab/openhab-core/pull/5090. You could potentially fall back to the "hack" if the `NetUtil` method can't resolve it - I think in the vast majority of cases it would, because multicast routing isn't easy to get working, and since both mDNS and the bridge communication depends on multicast, chances are very high that the address isn't on a routed network.

Reply:
That's an old hack. 😉 Comments appreciated, I have refactored it slightly now with cleaner resource handling, avoiding real connection attempt, and to work even if gateway is offline.